index.bs

```diff
@@ -5638,6 +5638,93 @@ At this time, one [=credential property=] is defined: the [=resident key credent
+
+Note: this extension may be implemented for [=authenticators=] that do not use [[FIDO-CTAP]] so long as the behavior observed by a [=[RP]=] is identical.
+
+: Extension identifier
+:: `prf`
+
+: Operation applicability
+:: [=registration extension|Registration=] and [=authentication extension|authentication=]
+
+: Client extension input ([=registration extension|registration=])
+::  <xmp class="idl">
+    partial dictionary AuthenticationExtensionsClientInputs {
+        boolean prfEnable;
+    };
+    </xmp>
+
+    <div dfn-type="dict-member" dfn-for="AuthenticationExtensionsClientInputs">
+        :   <dfn>prfEnable</dfn>
+        ::  If true, specifies that two PRFs should be associated with the created credential.
+    </div>
+
+: Client extension processing ([=registration extension|registration=])
+::
+       1. If {{AuthenticationExtensionsClientInputs/prfEnable}} is present and has the value `true`:
+           1. Set `hmac-secret` to `true` in the authenticator extensions input.
+           1. Set {{AuthenticationExtensionsClientOutputs/prfEnabled}} to the value of `hmac-secret` in the authenticator extensions output. If not present, set {{AuthenticationExtensionsClientOutputs/prfEnabled}} to false.
+       1. Else:
+           1. Set {{AuthenticationExtensionsClientOutputs/prfEnabled}} to false.
+
+: Client extension output ([=registration extension|registration=])
+:: <xmp class="idl">
+    partial dictionary AuthenticationExtensionsClientOutputs {
+        boolean prfEnabled;
+    };
+    </xmp>
+
+    <div dfn-type="dict-member" dfn-for="AuthenticationExtensionsClientOutputs">
+        :   <dfn>prfEnabled</dfn>
+        ::  True iff the two PRFs are available for use with the created credential.
+    </div>
+
+Note: it is not possible to evaluate a PRF during [=registration=] due to the design of the [[FIDO-CTAP]] `hmac-secret` extension.
+
+: Client extension input ([=authentication extension|authentication=])
+::  <xmp class="idl">
+    partial dictionary AuthenticationExtensionsClientInputs {
+        sequence<DOMString> prfEval;
+    };
+    </xmp>
+
+    <div dfn-type="dict-member" dfn-for="AuthenticationExtensionsClientInputs">
+        :   <dfn>prfEnable</dfn>
+        ::  A sequence of one or two strings that are inputs on which to evaluate the PRF.
+    </div>
+
+: Client extension processing ([=authentication extension|authentication=])
+::
+       1. If {{AuthenticationExtensionsClientInputs/prfEval}} is present and non-empty:
+           1. If the length of {{AuthenticationExtensionsClientInputs/prfEval}} is greater than two, return a {{DOMException}} whose name is “{{NotSupportedError}}”.
+           1. Let `salt1` be the value of <kbd>SHA-256(UTF8Encode("WebAuthn PRF\x00" + {{AuthenticationExtensionsClientInputs/prfEval}}[0]))</kbd>.
```

Comment:
If, in the future, we did v2 of this, one might otherwise do "WebAuthn PRF2" + input. However, that's then potentially ambiguous with an input that previously started with a 2. The NUL byte ensures that doesn't happen.